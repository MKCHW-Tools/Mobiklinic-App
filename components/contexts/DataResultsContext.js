import React, { createContext, useState } from 'react';

const DataResultsContext = createContext();

export const DataResultsProvider = ({ children }) => {
  const [dataResults, setDataResults] = useState('');
  const [benData, setBenData] = useState([]);
  const [userLog, setUserLog] = useState('');
  const [patientId, setPatientId] = useState('');
  const [userNames, setUserNames] = useState('');


  const updateDataResults = newDataResults => {
    setDataResults(newDataResults);
  };

  const updateBenData = newBenData => {
    setBenData(newBenData);
  };

  const updateUserLog = newUserLog => {
    setUserLog(newUserLog);
  };

  const updatePatientVac = newPatientVac => {
    setPatientVac(newPatientVac);
  };

  const updateUserNames = newUserNames => {
    setUserNames(newUserNames);
  };

  return (
    <DataResultsContext.Provider
      value={{
        dataResults,
        updateDataResults,
        benData,
        updateBenData,
        userLog,
        updateUserLog,
        patientId,
        setPatientId,
        userNames,
        updateUserNames
      }}
    >
      {children}
    </DataResultsContext.Provider>
  );
};

export default DataResultsContext;