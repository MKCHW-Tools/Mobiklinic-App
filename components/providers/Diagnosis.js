import * as React from 'react';

const DiagnosisContext = React.createContext();

const DiagnosisProvider = ({children}) => {
  const [followups, setFollowups] = React.useState([]);
  const [followup, setFollowup] = React.useState({});
  const [diagnoses, setDiagnoses] = React.useState([]);
  const [diagnosis, setDiagnosis] = React.useState({});

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const initial = React.useMemo(() => ({
    handleFollowup: () => {},
    handleFollowups: () => {},
    handleDiagnoses: async data => {},
    handleDiagnosis: () => {},
  }));

  return (
    <DiagnosisContext.Provider
      value={{
        diagnoses,
        followups,
        setDiagnoses,
        setFollowups,
      }}>
      {children}
    </DiagnosisContext.Provider>
  );
};

export {DiagnosisContext, DiagnosisProvider};
