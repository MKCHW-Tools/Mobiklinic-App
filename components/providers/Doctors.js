import * as React from 'react';

const DoctorsContext = React.createContext();

const DoctorsProvider = ({children}) => {
  const [doctors, setDoctors] = React.useState([]);

  return (
    <DoctorsContext.Provider
      value={{
        doctors,
        setDoctors,
      }}>
      {children}
    </DoctorsContext.Provider>
  );
};

export {DoctorsContext, DoctorsProvider};
