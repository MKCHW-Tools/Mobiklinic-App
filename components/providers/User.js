import * as React from "react";

const UserContext = React.createContext();

const UserProvider = ({ children }) => {
	const [accessToken, setAccessToken] = React.useState("");
	const [user, setUser] = React.useState({});
	const [isRegistering, setIsRegistering] = React.useState("no");

	return (
		<UserContext.Provider
			value={{
				accessToken,
				isRegistering,
				user,
				setUser,
				setAccessToken,
				setIsRegistering,
			}}>
			{children}
		</UserContext.Provider>
	);
};

export { UserProvider, UserContext };
