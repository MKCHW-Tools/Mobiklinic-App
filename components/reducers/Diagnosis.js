
export const initialState = {
    diagnoses:[],
    diagnosis:{},
    followup:{}
}

export const reducer = (state, action) => {
    switch(action.type) {
        case 'NEW_DIAGNOSIS':
            return {
                ...state,
                diagnosis
            }
        case 'FOLLOW_UP':
            return {
                ...state,
                followup
            }
        case 'DIAGNOSES':
            return {
                ...state,
                diagnoses
            }
        default:
            return state
    }
}