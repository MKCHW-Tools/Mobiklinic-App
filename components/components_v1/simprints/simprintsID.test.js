
import React from 'react';
import { render } from '@testing-library/react-native';
import { DiagnosisContext } from '../../providers/Diagnosis';
import DataResultsContext from '../../contexts/DataResultsContext';
import SimprintsID from '../../simprints/simprintsID';
import fetchMock from 'jest-fetch-mock';

// mock fetch
fetchMock.enableMocks();

describe('SimprintsID Component', () => {

    beforeEach(() => {
        fetchMock.resetMocks(); // Reset fetch mocks before each test
    });

    const navigation = {
        navigate: jest.fn(),
        goBack: jest.fn()
    };


    //Test that component renders correctly
    it('renders correctly', () => {

        const screen = render(
            <DiagnosisContext.Provider value={{ diagnoses: [] }}>
                <DataResultsContext.Provider value={{ dataResults: '', userLog: 'userLog', patientId: '', benData: [], setPatientId: jest.fn() }}>
                    <SimprintsID navigation={navigation} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>);

        expect(screen).toMatchSnapshot();

    });


    //Test for fetchData Function:
    it('fetchData fetches data and sets state', async () => {
        // Mock the fetch function to return a response
        fetchMock.mockResponseOnce(JSON.stringify({ id: 1 }), { status: 200 });


        const screen = render(
            <DiagnosisContext.Provider value={{ diagnoses: [] }}>
                <DataResultsContext.Provider value={{ dataResults: '', userLog: 'userLog', patientId: '', benData: [], setPatientId: jest.fn() }}>
                    <SimprintsID navigation={navigation} />
                </DataResultsContext.Provider>
            </DiagnosisContext.Provider>)

        // Wait for the fetchData function to be called and complete
        expect(fetchMock).toHaveBeenCalledTimes(1);

    });



});