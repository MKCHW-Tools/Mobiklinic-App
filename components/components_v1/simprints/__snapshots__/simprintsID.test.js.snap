// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimprintsID Component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255,0.6)",
      "flex": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "borderStyle": "solid",
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "height": 35,
          "justifyContent": "center",
          "marginHorizontal": 4,
          "opacity": 1,
          "width": 35,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#000",
              "fontSize": 25,
            },
            undefined,
            {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        [
          {
            "alignItems": "center",
            "color": "#000",
            "flex": 2,
            "fontSize": 16,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Biometrics Identification
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 2,
        "marginVertical": 30,
        "padding": 20,
      }
    }
  >
    <Image
      source=""
      style={
        {
          "height": 70,
          "width": 70,
        }
      }
    />
    <Text
      style={
        {
          "alignItems": "center",
          "color": "#7cd2ff",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginVertical": 20,
        }
      }
    >
      Mobiklinic
    </Text>
    <RCTScrollView>
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "flex": 1,
              "flexGrow": 1,
              "height": "100%",
              "marginVertical": 40,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#000",
                "borderRadius": 10,
                "marginBottom": 10,
                "marginTop": 10,
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Start Registration
            </Text>
          </View>
          <View
            style={
              {
                "height": 10,
              }
            }
          />
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#000",
                "borderRadius": 10,
                "marginBottom": 10,
                "marginTop": 10,
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#fff",
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              Identify Beneficiary
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexGrow": 1,
          "height": "100%",
          "marginVertical": 40,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#888",
              "fontSize": 12,
              "marginTop": 80,
              "textAlign": "center",
            }
          }
        >
          © 
          2023
           Mobiklinic. All rights reserved.
        </Text>
      </View>
    </View>
  </View>
</View>
`;
